client/store/reports/revenue/stats/reducer.js

```diff
@@ -0,0 +1,48 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { ERROR } from 'store/constants';
+
+const DEFAULT_STATE = {
+	queries: {},
+};
+
+export default function reportRevenueStatsReducer( state = DEFAULT_STATE, action ) {
+	if ( 'SET_REPORT_REVENUE_STATS' === action.type ) {
+		const prevQueries = get( state, 'queries', {} );
+		const query = JSON.stringify( action.query );
```

Comment:
Oh, one more small thing, you'll need to guarantee equal objects are serialised with properties in the same order:

```js
const query = JSON.stringify( action.query, Object.keys( action.query ).sort() );
```